pandas/core/indexing.py

```diff
@@ -1426,14 +1427,231 @@ class _LocIndexer(_LocationIndexer):
     - A list or array of labels, e.g. ``['a', 'b', 'c']``.
     - A slice object with labels, e.g. ``'a':'f'`` (note that contrary
       to usual python slices, **both** the start and the stop are included!).
-    - A boolean array.
+    - A boolean array of the same length as the axis being sliced,
+      e.g. ``[True, False, True]``.
     - A ``callable`` function with one argument (the calling Series, DataFrame
       or Panel) and that returns valid output for indexing (one of the above)
 
-    ``.loc`` will raise a ``KeyError`` when the items are not found.
-
     See more at :ref:`Selection by Label <indexing.label>`
 
+    See Also
+    --------
+    DateFrame.at : Access a single value for a row/column label pair
+    DateFrame.iloc : Access group of rows and columns by integer position(s)
+    Series.loc : Access group of values using labels
+
+    Examples
+    --------
+    **Getting values**
+
+    >>> df = pd.DataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+    ...      index=['r0', 'r1', 'r2'], columns=['c0', 'c1', 'c2'])
+    >>> df
+        c0  c1  c2
+    r0   1   2   3
+    r1   4   5   6
+    r2   7   8   9
+
+    Single label. Note this returns the row as a Series.
+
+    >>> df.loc['r1']
+    c0    4
+    c1    5
+    c2    6
+    Name: r1, dtype: int64
+
+    List with a single label. Note using ``[[]]`` returns a DataFrame.
+
+    >>> df.loc[['r1']]
+        c0  c1  c2
+    r1   4   5   6
+
+    Single label for row and column
+
+    >>> df.loc['r0', 'c1']
+    2
+
+    A list of labels
+
+    >>> df.loc[['r1', 'r2']]
+        c0  c1  c2
+    r1   4   5   6
+    r2   7   8   9
+
+    Slice with labels for row and single label for column. Note that
+    contrary to usual python slices, both the start and the stop are
+    included!
+
+    >>> df.loc['r0':'r1', 'c0']
+    r0     1
+    r1     4
+    Name: c0, dtype: int64
+
+    Boolean list with the same length as the row axis
+
+    >>> df.loc[[False, False, True]]
```

Comment:
this is not a very common thing to do (directly), the boolean indexing right below is MUCH more important.